pkg/plugins/policies/core/xds/meshroute/clusters.go

```diff
@@ -117,16 +119,30 @@ func GenerateClusters(
 			}
 
 			resources = resources.Add(&core_xds.Resource{
-				Name:     clusterName,
-				Origin:   generator.OriginOutbound,
-				Resource: edsCluster,
+				Name:           clusterName,
+				Origin:         generator.OriginOutbound,
+				Resource:       edsCluster,
+				ResourceOrigin: createOwner(service.BackendRef(), meshCtx),
+				Protocol:       protocol,
 			})
 		}
 	}
 
 	return resources, nil
 }
 
+func createOwner(ref common_api.BackendRef, meshCtx xds_context.MeshContext) *core_rules.UniqueResourceIdentifier {
+	if ref.Kind == common_api.MeshService && ref.ReferencesRealObject() {
```

Comment:
something is odd here, `ref.ReferencesRealObjects()` checks if `ref.Kind` is equal to `MeshService`

Reply:
yes, but I don't want to implement this for other kinds for now, I just wanted to handle MeshService, also other resource are in different parts of meshCtx

Reply:
But if you know that `ref.Kind` is `MeshService` do you need to call `ref.ReferencesRealObjects()`?

Reply:
I think so, we do this in other planes route: https://github.com/kumahq/kuma/blob/982dbdc6819c899f0c4854b5e5dac07942ee8a91/pkg/plugins/policies/core/xds/meshroute/listeners.go#L241

Reply:
I think there could be old backendRef with MeshService kind, but old one